cli/src/commands/git/push.rs

```diff
@@ -418,8 +463,72 @@ fn validate_commits_ready_to_push(
             }
             return Err(error);
         }
+        if commit.is_signed() != sign_settings.should_sign(commit.store_commit()) {
+            commits_to_sign.push(commit);
+        }
     }
-    Ok(())
+    Ok(commits_to_sign)
+}
+
+/// Signs commits before pushing if configured to do so.
+///
+/// Returns the number of commits with rebased descendants and the updated list
+/// of bookmark names and corresponding [`BookmarkPushUpdate`]s.
+fn sign_commits_before_push(
+    tx: &mut WorkspaceCommandTransaction,
+    bookmark_updates: Vec<(String, BookmarkPushUpdate)>,
+    commits_to_sign: Vec<Commit>,
+    dry_run: bool,
+) -> Result<(usize, Vec<(String, BookmarkPushUpdate)>), CommandError> {
+    let sign_behavior = if tx.settings().get_bool("git.sign-on-push").unwrap_or(false) {
+        SignBehavior::Own
+    } else {
+        SignBehavior::Keep
```

Comment:
When we're just keeping signatures, what's the point of even transforming them? I.e. can. we skip calling `sign_commits_before_push()` completely when `git.sign-on-push` is off?